GVFS/GVFS.Platform.POSIX/POSIXPlatform.cs

```diff
@@ -7,19 +7,24 @@
 using System.IO;
 using System.IO.Pipes;
 using System.Runtime.InteropServices;
-using System.Security;
-
+using System.Security;
+
 namespace GVFS.Platform.POSIX
 {
     public abstract partial class POSIXPlatform : GVFSPlatform
     {
-        public POSIXPlatform() : base(
+        protected POSIXPlatform() : this(
             underConstruction: new UnderConstructionFlags(
                 supportsGVFSUpgrade: false,
                 supportsGVFSConfig: false))
         {
         }
 
+        protected POSIXPlatform(UnderConstructionFlags underConstruction)

```

Comment:
It's not apparent to me from this PR or the description, why does `POSIXPlatform` not support config?

Reply:
I implemented this at the POSIX layer, but I hadn't tested it on a Linux platform. I thought this was a good middle ground between implement at the common layer, but enable at the layer that is tested.

Also, the overarching change will include upgrader related config settings, which also will not be tested on Linux at initially.